MonoGame.Framework/Platform/Native/EffectResource.Native.cs

```diff
@@ -21,8 +21,10 @@ private static unsafe byte[] PlatformGetBytecode(string name)
     {
         byte* data;
         int size;
-        fixed (byte* n = System.Text.Encoding.UTF8.GetBytes(name + '\0'))
-            MGG.EffectResource_GetBytecode(n, &data, &size);
+
+        byte* _name = stackalloc byte[StringInterop.GetMaxSize(name)];
+        StringInterop.CopyString(_name, name);
+        MGG.EffectResource_GetBytecode(_name, out data, out size);
```

Comment:
This removes a heap allocation when interop of strings.  Uses a guess of `string.Length * 4` for the length so we don't have to convert the string twice to copy it.